### DISCUSSION TITLE
Plugin "QuickLibrary" for quick insert of visual elements

### DISCUSSION THREAD

**Discussion by RaddishHorse** [xournalpp/xournalpp#5525] [Show and tell]
Hello, 

I developed in the past few days a small plugin for Xournalpp (tested on 1.2.2, Linux, Gtk3), which opens a window to quickly insert visual elements (images) into the canvas.

It would be great to get some feedback or comments! 

**Download and sources are available here:**
[Link to the repository](https://github.com/RaddishHorse/QuickLibrary)

**Screenshot:**
[screenshot.png](https://github.com/RaddishHorse/QuickLibrary/raw/main/screenshot.png)

**Comment by tattsan** [comment]
Interesting!
Is it possible to develop the library to include other xournalpp objects, such as handwritten diagrams?

**Reply by RaddishHorse** [reply]
Thanks for that feedback and idea.

Not sure how to do that with the current Lua API. As far as I can see, only Pixbuf images can be imported, and no other object nor SVG can be either exported or imported. 

As the workflow, I could totally imagine selecting the objects, importing into the library (saved as file on disk, either as a xopp object or SVG), and then being importable and again modifiable into the canvas.

Maybe somebody more knowledgeable about it could weigh in?

**Reply by rolandlo** [reply]
You could use `app.addStrokes` and `app.getStrokes` for adding strokes to the document/getting strokes from the document. It's not possible to import the contents of a xopp-file directly though. So either the strokes would have to be saved in a custom format or the plugin would have to read the strokes itself from a xopp-file or the Lua API would have to be extended. Probably using a custom format that is close to what needs to be passed to `app.addStrokes` (or what is returned from `app.getStrokes` is the solution I would go with.

**Reply by rolandlo** [reply]
I think the plugin should also work on Windows. See [this recipe](https://github.com/xournalpp/xournalpp/discussions/4522#discussioncomment-8789465) to install lgi on Windows.
